src/status_im/ui/screens/communities/views.cljs

```diff
@@ -0,0 +1,523 @@
+                                              :accessibility-label :community-name-text
+                                              :ellipsize-mode      :tail
+                                              :number-of-lines     1}
+                                    (utils/truncate-str (:display-name identity) 30)]]]
+      :title-accessibility-label :community-name-text
+      :subtitle                  [react/view {:flex-direction :row}
+                                  [react/view {:flex 1}
+                                   [quo/text
+                                    (utils/truncate-str (:description identity) 30)]]]
+      :on-press                  #(do
+                                    (re-frame/dispatch [:dismiss-keyboard])
+                                    (re-frame/dispatch [:navigate-to :community id]))}]))
+
+(defn communities-actions []
+  [react/view
+   [quo/list-item
+    {:theme               :accent
+     :title               (i18n/label :t/import-community)
+     :accessibility-label :community-import-community
+     :icon                :main-icons/check
+     :on-press            #(hide-sheet-and-dispatch [::import-pressed])}]
+   [quo/list-item
+    {:theme               :accent
+     :title               (i18n/label :t/create-community)
+     :accessibility-label :community-create-community
+     :icon                :main-icons/check
+     :on-press            #(hide-sheet-and-dispatch [::create-pressed])}]])
+
+(views/defview communities []
+  (views/letsubs [communities [:communities]]
+    [react/view {:flex 1}
+     [topbar/topbar (cond-> {:title (i18n/label :t/communities)}
+                      config/communities-management-enabled?
+                      (assoc :right-accessories [{:icon                :main-icons/more
+                                                  :accessibility-label :chat-menu-button
+                                                  :on-press
+                                                  #(re-frame/dispatch [:bottom-sheet/show-sheet
+                                                                       {:content (fn []
+                                                                                   [communities-actions])
+                                                                        :height  256}])}]))]
+     [react/scroll-view {:style                   {:flex 1}
+                         :content-container-style {:padding-vertical 8}}
+      [list/flat-list
+       {:key-fn                       :id
+        :keyboard-should-persist-taps :always
+        :data                         (vals communities)
+        :render-fn                    (fn [community] [community-list-item community])}]]
+     (when config/communities-management-enabled?
+       [toolbar/toolbar
+        {:show-border? true
+         :center [quo/button {:on-press #(re-frame/dispatch [::create-pressed])}
+                  (i18n/label :t/create)]}])]))
+
+(fx/defn import-pressed
```

Comment:
why are events in the ui ns ?

Reply:
They seem to be very ui specific, basically if I put them on say `communities/core.cljs` then that namespace will have to import some ui components, and that will lead to circular dependencies (potentially), as views should import models and not the other way around.
I could put them in `src/status_im/events.cljs`, that would be acceptable, though I almost feel they loose visibility (I'd have to jump files to see them, and is a large file).
I don't mind them here as they are very UI-centric `import-pressed`, `create-pressed`.
But not sure what's best :)
Where do you think is the most appropriate place?

Reply:
we have lots of very ui specific events in different core ns, thats how we usually do it, put all events in core, ui events have ui namespace in the key , i feel like its better trying to keep the same patterns in the whole code base, overwise it will be a mess imo